docs/recipes/WritingTests.md

```diff
@@ -79,7 +79,7 @@ export default function todos(state = initialState, action) {
   switch (action.type) {
   case ADD_TODO:
     return [{
-      id: state.length,
+      id: state.reduce((maxId, todo) => Math.max(todo.id, maxId), -1) + 1,
```

Comment:
Hi @gaearon, Hi @acdlite,

I would like to know that how this particular line of code works.
I am struggling to understand that from where **maxId** has come and how the ids are getting generated.

Please help me with this. I am new to redux and wish to use it in my current project.

Thanks in advance.